handler.js

```diff
@@ -166,39 +162,39 @@ function approve(responseUrl, emailAtt, user) {
 
   console.log(`Sending to ${sendTo}`);
 
-  ses.sendEmail({
-    Source: process.env.SEND_FROM,
-    Destination: {
-      ToAddresses: sendTo
-    },
-    ReplyToAddresses: [tmplContext.author_name],
-    Message: {
-      Subject: {
-        Data: emailAtt.title
+  try {
+    await ses.sendEmail({
+      Source: process.env.SEND_FROM,
+      Destination: {
+        ToAddresses: sendTo
       },
-      Body: {
-        Text: {
-          Data: EMAIL_TEMPLATE(tmplContext),
-          Charset: 'UTF-8'
+      ReplyToAddresses: [tmplContext.author_name],
+      Message: {
+        Subject: {
+          Data: emailAtt.title
+        },
+        Body: {
+          Text: {
+            Data: EMAIL_TEMPLATE(tmplContext),
+            Charset: 'UTF-8'
+          }
         }
       }
-    }
-  }, (err, _data) => {
-    if (err) {
-      return errorToSlack(responseUrl, err);
-    }
+    }).promise();
```

Comment:
The intent of this line is obvious (convert whatever `sendEmail` returns into a promise) but just for my own interest I'd like to know more about the `.promise()` method. Am I correct in thinking that it's the `.promise()` method that Amazon provides as mentioned [here](https://aws.amazon.com/blogs/developer/support-for-promises-in-the-sdk/)?

Reply:
That's correct.